File: Creational/FactoryMethod/Java/products/WinButton.java
package patterns.creational.factorymethod.products;

import patterns.creational.factorymethod.interfaces.IButton;

/**
 * Concrete product.
 */

public class WinButton implements IButton
{
	/**
	 * @Overrride
	 */
	public void paint()
	{
		System.out.println("Painting button in Windows style.");
	}

	/**
	 * @Override
	 */
	public void setPosition(final int x, final int y)
	{
		System.out.println("Setting button position to (" + x + ", " + y + ") in Windows style.");
	}
}

File: Creational/FactoryMethod/C++/products/WinButton.cpp
#include <iostream>
#include "WinButton.h"

/**
 * Concrete product (realisation).
 */

void WinButton::paint()
{
	std::cout << "Painting button in Windows style." << std::endl;
}

void WinButton::setPosition(const int & x, const int & y)
{
	std::cout << "Setting button position to (" << x << ", " << y << ") in Windows style." << std::endl;
}
